File: backend/app/schemas/session.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict


class SessionCreate(BaseModel):
    title: Optional[str] = None
    user_id: Optional[str] = None


class SessionResponse(BaseModel):
    id: str
    user_id: Optional[str] = None
    title: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SessionListResponse(BaseModel):
    sessions: List[SessionResponse]

File: frontend/src/types/session.ts
export interface Session {
  id: string;
  user_id?: string | null;
  title?: string | null;
  created_at: string;
  updated_at: string;
}

export interface SessionListResponse {
  sessions: Session[];
}

export interface SessionCreatePayload {
  title?: string;
  user_id?: string;
}
